modules/virt-preparing-container-disk-for-vms.adoc

```diff
@@ -0,0 +1,50 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/virt-create-vms.adoc
+
+[id="virt-preparing-container-disk-for-vms_{context}"]
+= Preparing a container disk for virtual machines
+
+A container disk must be built with a virtual machine image and pushed to a container registry. It can then either be imported into a DataVolume and attached to a virtual machine or attached directly to a virtual machine as an ephemeral containerDisk volume.
+
+.Prerequisites
+
+* You have root or sudo privileges on your local system to install podman.
+
+* The virtual machine image must be either QCOW2 or RAW format.
+
+.Procedure
+
+. Install `podman` if it is not already installed:
++
+[source,terminal]
+----
+$ sudo yum install podman -y
+----
+
+. Create a Dockerfile to build the virtual machine image into a container image.
+The virtual machine image must be owned by QEMU, which has a UID of `107`, and placed in the `/disk` directory inside the container.
++
+[source,terminal]
+----
+$ cat > Dockerfile << EOF
+FROM scratch
+ADD --chown=107:107 <vm_image>.qcow2 /disk/ <1>
+EOF
+----
```

Comment:
In order to set the 440 file mode, we can use a multi-stage docker build. 

here's an example that uses the standard ubi container image as the builder image, then uses a scratch image to store the result. 

```
FROM registry.access.redhat.com/ubi8/ubi:latest AS builder
ADD --chown=107:107 Fedora-Cloud-Base-32-1.6.x86_64.qcow2 /disk/
RUN chmod 0440 /disk/*

FROM scratch
COPY --from=builder /disk/* /disk/
```